### ISSUE TITLE
Use terminal colors in theme

### ISSUE THREAD

**Issue by cbarrete** [mfontanini/presenterm#64]
Hi, is there a way to get a theme to use the terminal's colors (background, foreground, primary colors)?

The default themes look fine but:
* I'd love to be able to reuse the colors that I've already set in my terminal
* The margins around the presentation, which use my terminal's background are a bit jarring (blue against black)

If so, it would also be nice to have a built-in theme that uses them.

I'm happy to contribute (especially with a few pointers) if you agree that this would be desirable.

**Comment by cbarrete** [comment]
![Contrived example](https://github.com/mfontanini/presenterm/assets/62146989/fd8c07a7-2dd1-4c30-84a4-51e49a09cebe)

**Comment by mfontanini** [comment]
I think it may be easier to just define the theme yourself and select which colors you want for that. The background and foreground ones are easy, but the primary colors would be trickier as you probably want them in specific contexts (e.g. _that_ color in the slide title). This is something you need to define once and never again and it will look exactly how you want it to.

**Comment by cbarrete** [comment]
Yeah, that's definitely possible, but it's tedious, and it means each user would have to manually pick out the colors and convert them to the right yaml format. Not to mention that this has to be redone when someone changes their terminal colors.

Regarding the "_that_ color in the slide title", I personally don't care what specific color is used for a specific element, and I assume many others others don't either, but using the primary colors for the main elements (title, headers, `code`, etc.) couldn't go particularly wrong? Either way, that's an argument against having that theme be built-in, not having access to terminal colors when writing a theme.

**Comment by mfontanini** [comment]
Fair, I think allowing this to be used but not having any built-in themes that use it makes sense.

**Comment by weygoldt** [comment]
Coincidentally, I am looking for the same behavior just now and would second @cbarrete. I think your project is great and is currently the only terminal presentation viewer with modern image support and active development. But I would use terminal presentations mainly when I want to show stuff in the terminal as well, meaning that I would switch between a presentation and a terminal window all the time. Not having to manually adjust the colors of the slides every time I switch terminal theme would be a huge plus!

**Comment by mfontanini** [comment]
Nice, that's a really good point. So just to be clear, there would be colors like "black" or "red" that would be mapped by your terminal and I guess there could be a "terminal" theme that defaults foreground/background color to your terminal's and would use some of these colors for the markdown elements that have some (e.g. slide titles, headers, etc). I'm guessing there should be 2 variants (dark/light) to this theme so that the code syntax highlighting doesn't look odd. e.g. I don't want a light highlighting theme on my dark themed terminal.

Does this make sense?

**Comment by cbarrete** [comment]
Yes, that's exactly what I had in mind!

**Comment by weygoldt** [comment]
@mfontanini Yes that's exactly what I would look for. I often use a python package called `rich` that can print rich text to the terminal, including markdown files and automatically uses the terminal colors for that and does a great job. In the case of slides I would even go as far as completely disabling the background color for that theme (if that is possible). Some people (like me) have a little transparency on their terminal background and a non-transparent slide background looks off. And for the other people that have no transparency, not having a set background color does not change anything as the terminal background color is already the theme background color.

**Comment by mfontanini** [comment]
Sweet! I'll look into this next. I _think_ you may be able to set "no background" already by setting it to `null` on the theme. I presume that's how yaml represents this given colors already are an `Option`. But I'll make sure it works, add support for primary colors, and add the light/dark theme.

**Comment by mfontanini** [comment]
> I think you may be able to set "no background" already by setting it to null on the theme

This sort of works but gets screwed up with code blocks so it's not there yet.

**Comment by mfontanini** [comment]
I just pushed this. There's 2 new `terminal-dark` and `terminal-light` themes. Let me know if anything looks off. This is the dark one [![asciicast](https://asciinema.org/a/U01vpgMNc4Qh4GGPYokw0gAcx.svg)](https://asciinema.org/a/U01vpgMNc4Qh4GGPYokw0gAcx)

Btw I've been thinking of adding a config files under `~/.config` or something so that you can define the theme globally rather than doing this on a per presentation basis (or when invoking the tool). Would that work for you?

**Comment by weygoldt** [comment]
Wow, that was quick, thank you so much! I'll make sure to test the themes tomorrow. Yes, having this as a default for all presentations somewhere in the dotfiles would be a great idea I think!

**Comment by mfontanini** [comment]
Awesome, glad it worked!

**Comment by cbarrete** [comment]
Yes it's great, thanks for getting it working so quickly!

**Comment by mfontanini** [comment]
Currently on master you can now drop highlighting themes under `~/.config/presenterm/themes` and they will be loaded on startup and considered built-in so you can `theme: potato` in the front matter or `--theme potato` when running the tool. 

Also after #74 you can drop a config.yaml file in your `~/.config` to set the default theme to be used for all presentations. See the PR for more info.
